tests/test_jwt.py

```diff
@@ -28,6 +34,19 @@ def test_non_default_alg(self, claims, key):
         decoded = jwt.decode(encoded, key, algorithms='HS384')
         assert claims == decoded
 
+    def test_non_default_alg_positional_bwcompat(self, claims, key):
+        encoded = jwt.encode(claims, key, 'HS384')
+        decoded = jwt.decode(encoded, key, 'HS384')
+        assert claims == decoded
+
+    def test_non_default_headers(self, claims, key, headers):
+        encoded = jwt.encode(claims, key, headers=headers)
+        decoded = jwt.decode(encoded, key)
+        assert claims == decoded
+        all_headers = jwt.get_unverified_headers(encoded)
+        custom_headers = {k: all_headers[k] for k in headers.keys()}
```

Comment:
Python 2.6 doesn't support dict comprehension so you will probably have to do this the longer way.  Something like this should work.

```python
for k, v in headers.items():
    assert all_headers[k] == v
```